Docs: add nx-neptune backend documentation

acarbonetto opened the pull request:
Adds nx-neptune to backend.md list of compatible integration clients. 

References:
* https://pypi.org/project/nx-neptune/
* https://github.com/awslabs/nx-neptune
* Discussion: https://github.com/networkx/networkx/discussions/6079
* Example notebooks: https://github.com/awslabs/nx-neptune?tab=readme-ov-file#jupyter-notebook-integration

## Install from pypi: 
```
pip install nx_neptune
```

Some notable changes: 
* Backend documentation is updated with a link to the nx-neptune repository. [Example](https://output.circle-artifacts.com/output/job/9f31fd5a-3461-42e6-a769-a4252139f93d/artifacts/0/doc/build/html/backends.html) 
* Supported algorithms contain links to the nx-neptune repository with additional information, such as links to source and the Amazon documentation page, as well as the API.  See [algorithms.md](https://github.com/awslabs/nx-neptune/blob/main/algorithms.md). 

<img width="913" height="232" alt="image" src="https://github.com/user-attachments/assets/b5b1d097-6d18-4d2e-ab16-84ead5434c98" />
<img width="701" height="195" alt="image" src="https://github.com/user-attachments/assets/f83478e9-a1aa-4390-b156-2da473b449bc" />

MridulS commented:
@acarbonetto thanks for this! It's nice to see more backends implemented in the wild :)

NetworkX has weekly community meetings and I think it would be great to get a quick demo of this if possible! We meet 9am Vancouver time weekly on Thursdays. Link to the .ics https://scientific-python.org/calendars/networkx.ics. It's okay if you can't make it, no pressure :) I would just need to go fiddle around with AWS then.

acarbonetto commented:
@rossbar @Schefflera-Arboricola 

I have updated our documentation (no change required here) on the nx-neptune side of things and moved a majority of the API documentation to the nx-neptune site - only a single click away.  

All information the algorithms are consolidated in our [algorithms.md](https://github.com/awslabs/nx-neptune/blob/main/algorithms.md) file with anchor links from the networkx.org documentation page.  

For example, for `bfs_edges`, it's just a link:  https://output.circle-artifacts.com/output/job/9f31fd5a-3461-42e6-a769-a4252139f93d/artifacts/0/doc/build/html/reference/algorithms/generated/networkx.algorithms.traversal.breadth_first_search.bfs_edges.html#networkx.algorithms.traversal.breadth_first_search.bfs_edges
While for 
While for `asyn_lpa_communities` there's a few more information, but most of the details are in the link: https://output.circle-artifacts.com/output/job/9f31fd5a-3461-42e6-a769-a4252139f93d/artifacts/0/doc/build/html/reference/algorithms/generated/networkx.algorithms.community.label_propagation.asyn_lpa_communities.html#networkx.algorithms.community.label_propagation.asyn_lpa_communities

Let me know if this is acceptable.

acarbonetto commented:
> By the way, the testing is set up so you set some environment variables and then run all the networkx test suite. For functions not provided by neptune get run in NetworkX while the provided functions (including when other networkx function call that function) use the neptune backend. In your case you still need to set up the resource to make it run. But you don't need to worry about which tests to run -- just run them all and rely on the fallback to allow the tests to pass.

Correct.  I can run these locally, as we don't yet have a set of resources to run against in CI.  This is partially why we are labelling the release as "alpha".  

> As far as timing goes, we plan to release networkx 3.6 in October. So the docs would become public then. It looks like nx-neptune is in an Alpha release phase based on the Readme. Do you want the NX public links to be available while in Alpha?

As mentioned during the demo, the other reason for the "alpha" label is to get usage data before a v1.0 release. The preview library is stable, within the test scenarios that we've run - but we'd like to get more pipelines and use cases before declaring it stable. 

Regardless, the backend is out in the wild and we want people to use it.

MridulS reviewed (approved):
Thanks @acarbonetto !
